k8s/src/test/scala/io/buoyant/k8s/IngressCacheTest.scala

```diff
@@ -1,12 +1,332 @@
+        }
+      },
+      "spec": {
+        "rules": [
+          {
+            "http": {
+              "paths": [
+                {
+                  "path": "/linkerd-only",
+                  "backend": {
+                    "serviceName": "echo2",
+                    "servicePort": 7070
+                  }
+                },
+                {
+                  "path": "/shared",
+                  "backend": {
+                    "serviceName": "echo",
+                    "servicePort": 7071
+                  }
+                }
+              ]
+            }
+          }
+        ]
+      },
+      "status": {
+        "loadBalancer": {}
+      }
+    }
+  ]
+}"""
+
+  val ingressResourceListWithOneIngress =
+    """
+{
+  "kind": "IngressList",
+  "apiVersion": "extensions/v1beta1",
+  "metadata": {
+    "selfLink": "/apis/extensions/v1beta1/ingresses",
+    "resourceVersion": "58845"
+  },
+  "items": [
+    {
+      "metadata": {
+        "name": "the-ingress",
+        "namespace": "default",
+        "selfLink": "/apis/extensions/v1beta1/namespaces/default/ingresses/the-ingress",
+        "uid": "6c1466d7-813e-11e7-b89b-080027a996b8",
+        "resourceVersion": "58840",
+        "generation": 1,
+        "creationTimestamp": "2017-08-14T22:17:55Z"
+      },
+      "spec": {
+        "rules": [
+          {
+            "http": {
+              "paths": [
+                {
+                  "path": "/some-path",
+                  "backend": {
+                    "serviceName": "echo",
+                    "servicePort": 1010
+                  }
+                },
+                {
+                  "path": "/other-path",
+                  "backend": {
+                    "serviceName": "echo",
+                    "servicePort": 2021
+                  }
+                }
+              ]
+            }
+          }
+        ]
+      },
+      "status": {
+        "loadBalancer": {}
+      }
+    }
+  ]
+}"""
+
+  val annotationClass = "linkerd"
+
+  def mkIngressApiServiceReturning(response: String) = Service.mk[Request, Response] {
+    case req if req.uri.contains("/apis/extensions/v1beta1/ingresses") =>
+      val rsp = Response()
+      rsp.content = Buf.Utf8(response)
+      Future.value(rsp)
+    case req =>
+      fail(s"unexpected request for [${req.uri}]: $req")
+  }
+
+  test("when multiple ingress resources, only adds one with desired annotation class") {
+    val service = mkIngressApiServiceReturning(ingressResourceListWithManyAnnotatedIngresses)
+    val cache = new IngressCache(None, service, annotationClass)
+    assert(await(cache.matchPath(host, "/istio-only")).isEmpty)
+    assert(await(cache.matchPath(host, "/nginx-only")).isEmpty)
+    assert(await(cache.matchPath(host, "/non-existing-path")).isEmpty)
+    assert(await(cache.matchPath(host, "/linkerd-only")).get.svc == "echo")
+  }
+
+  test("when multiple ingress resources with desired annotation class, only adds the first one") {
```

Comment:
This does not describe the behavior I expect to see. Linkerd will add as many ingress resources as have correct annotations. Whether linkerd chooses to match to echo1 or echo2 is undefined in this case